Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1655,3 +1655,66 @@ function Set-ToolsRepo
         }
     }
 }
+
+<#
+.Synopsis
+    Set vSAN compression and deduplication on a cluster or clusters. If deduplication is enabled then compression is required.
+    The default cluster configuration is deduplication and compression but the customer can change that.
+    Choosing neither compression nor deduplication will disable both.
+    This requires action on every physical disk and will take time to complete.
+.EXAMPLE
+    Set-vSANCompressDedupe -ClustersToChange "cluster-1,cluster-2" -Compression $true
+    Set-vSANCompressDedupe -ClustersToChange "cluster-1,cluster-2" -Deduplication $true
+    Set-vSANCompressDedupe -ClustersToChange "cluster-1,cluster-2"
+    Set-vSANCompressDedupe -ClustersToChange "*"
+#>
+function Set-vSANCompressDedupe
+{
+    [AVSAttribute(60, UpdatesSDDC = $true)]
+    param(
+        [Parameter(Mandatory = $true)]
+        [String]$ClustersToChange,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression and deduplication.")]
+        [bool]$Deduplication = $false,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression only.")]
+        [bool]$Compression = $false
+    )
+
+    # $cluster is an array of cluster names or "all"
+    If ($ClustersToChange -eq "*")
```

Comment:
With the wildcards there's no need to if/else branching - if user passes "*" the logic in else branch will handle it just fine.

Reply:
Good point. Changed